tsl/src/compression/compression_scankey.c

```diff
@@ -22,6 +22,29 @@ static int create_segment_filter_scankey(Relation in_rel, char *segment_filter_c
 										 Bitmapset **null_columns, Datum value, bool is_null_check,
 										 bool is_array_op);
 
+/*
+ * Test ScanKey against a slot.
+ *
+ * Unlike HeapKeyTest, this function takes into account SK_ISNULL
+ * and works correctly when looking for null values.
+ */
+bool
+slot_key_test(TupleTableSlot *compressed_slot, ScanKey key)
+{
+	Datum val;
+	bool is_null;
+	val = slot_getattr(compressed_slot, key->sk_attno, &is_null);
+
+	/* NULL values only match the value is NULL */
+	if (key->sk_flags & SK_ISNULL)
+		return is_null;
+
+	if (is_null)
+		return false;
```

Comment:
not sure we wanna complicate this but when we looking for NULL we only need to do slot_attisnull and not pull out datum